pennylane/circuit_drawer/representation_resolver.py

```diff
@@ -0,0 +1,406 @@
+                    )
+
+                    poly_str += RepresentationResolver._format_poly_term(
+                        coefficients[p1, p2] + coefficients[p2, p1],
+                        "p{}p{}".format(
+                            self.charset.to_subscript(idx1), self.charset.to_subscript(idx2)
+                        ),
+                    )
+
+                    poly_str += RepresentationResolver._format_poly_term(
+                        coefficients[x1, p2] + coefficients[p2, x1],
+                        "x{}p{}".format(
+                            self.charset.to_subscript(idx1), self.charset.to_subscript(idx2)
+                        ),
+                    )
+
+                    poly_str += RepresentationResolver._format_poly_term(
+                        coefficients[p1, x2] + coefficients[x2, p1],
+                        "x{}p{}".format(
+                            self.charset.to_subscript(idx2), self.charset.to_subscript(idx1)
+                        ),
+                    )
+
+        return poly_str
+
+    # pylint: disable=too-many-return-statements
+    def operator_representation(self, op, wire):
+        """Return the string representation of an Operator.
+
+        Args:
+            op (pennylane.operation.Operator): The Operator instance whose representation shall be returned
+            wire (int): The Operator's wire for which the string representation shall be returned
+
+        Returns:
+            str: String representation of the Operator
+        """
+        if isinstance(op, qml.operation.Tensor):
+            constituent_representations = [
+                self.operator_representation(tensor_obs, wire) for tensor_obs in op.obs
+            ]
+
+            return (" " + self.charset.OTIMES + " ").join(constituent_representations)
+
+        name = op.name
+
+        # Use a shorter name if applicable
+        if name in RepresentationResolver.resolution_dict:
+            name = RepresentationResolver.resolution_dict[name]
+
+        # Display a control symbol for all controlling qubits of a controlled Operation
+        if op.name in self.control_wire_dict and wire in [
+            op.wires[control_idx] for control_idx in self.control_wire_dict[op.name]
+        ]:
+            return self.charset.CONTROL
+
+        if op.num_params == 0:
+            return name
+
+        if op.name == "QubitUnitary":
+            return RepresentationResolver._format_matrix_operation(
+                op, "U", self.unitary_matrix_cache
+            )
+
+        if op.name == "Hermitian":
+            return RepresentationResolver._format_matrix_operation(
+                op, "H", self.hermitian_matrix_cache
+            )
+
+        if op.name == "QuadOperator":
+            par_rep = self.single_parameter_representation(op.params[0])
+
+            return "cos({0})x+sin({0})p".format(par_rep)
+
+        if op.name == "FockStateProjector":
+            n_str = ",".join([str(n) for n in op.params[0]])
```

Comment:
An idea here, since there are many ``if`` statements would be to create a ``dict``. This would have ``op.name`` as the key and the constructed ``string`` as the value. There are some values that are duplicated, but I guess it should be fine.

Then the return would look something like ``return control_representations[op.name]``.

Reply:
I don't see how that would work. If one were to convert this to a dict it would have to be a dict with a lot of lambdas in it -- and that's surely less readable.
```python
string_representations = {
    "Hermitian" : lambda op: RepresentationResolver._format_matrix_operation(
                op, "H", self.hermitian_matrix_cache
            ),
    "etc" : ...
}
```